src/sources/util/tcp.rs

```diff
@@ -190,7 +191,9 @@ fn handle_stream(
         }
 
         // Actual work
-        reader.poll()
+        let result = reader.poll();
+        let _ = token.take();
```

Comment:
Is this functionally any different than `drop(token)`?

Reply:
~`drop(token)` still produces error message~ tested again and no messages :thinking: 
Will change to drop.